code/modules/mob/living/simple_animal/hostile/carp.dm

```diff
@@ -59,15 +60,30 @@
 	icon_living = "megacarp"
 	icon_dead = "megacarp_dead"
 	icon_gib = "megacarp_gib"
-	maxHealth = 65
-	health = 65
+	maxHealth = 130
+	health = 130
 	pixel_x = -16
 	mob_size = MOB_SIZE_LARGE
 
 	obj_damage = 80
 	melee_damage_lower = 20
-	melee_damage_upper = 20
+	melee_damage_upper = 30
 
+	var/regen_cooldown = 0
+
+/mob/living/simple_animal/hostile/carp/megacarp/Initialize()
+	. = ..()
+	name = "[pick(GLOB.megacarp_first_names)] [pick(GLOB.megacarp_last_names)]"
+
+/mob/living/simple_animal/hostile/carp/megacarp/adjustHealth(amount, updating_health = TRUE, forced = FALSE)
+	. = ..()
+	if(.)

```

Comment:
this will trigger on each regeneration tick

Reply:
is that bad

Reply:
okay bad question, how about this
what should i replace it with